common/tier4_autoware_utils/include/tier4_autoware_utils/trajectory/trajectory.hpp

```diff
@@ -667,6 +667,41 @@ inline boost::optional<size_t> insertTargetPoint(
 
   return seg_idx;
 }
+
+/**
+ * @brief calculate the point offset from source point along the trajectory (or path)
+ * @param insert_point_length length to insert point from the beginning of the points
+ * @param p_target point to be inserted
+ * @param points output points of trajectory, path, ...
+ * @return index of insert point
+ */
+template <class T>
+inline boost::optional<size_t> insertTargetPoint(
+  const double insert_point_length, const geometry_msgs::msg::Point & p_target, T & points,
+  const double overlap_threshold = 1e-3)
```

Comment:
Don't you add specialized function for PathWithLaneId ?

Reply:
done